### PR TITLE
Sentry sourcemaps upload — now with better pathes

### PR DESCRIPTION
**Pull request by y-lohse**
I've copy-pasted some code from the internets.

### CODE REVIEW

**Comment by y-lohse** [comment]
[(1) 📱 Améliorer notre gestion de Sentry](https://trello.com/c/oaQKEI6x/811-1-%F0%9F%93%B1-am%C3%A9liorer-notre-gestion-de-sentry)

File: src/drive/mobile/lib/reporter.js [hunk 1]
```diff
@@ -6,11 +6,34 @@ export const ANALYTICS_URL =
     ? ''
     : `https://${__SENTRY_TOKEN__}@sentry.cozycloud.cc/6`
 
+// normalize files path on mobile, see https://github.com/getsentry/sentry-cordova/blob/17e8b3395e8ce391ecf28658d0487b97487bb509/src/js/SentryCordova.ts#L213
+const normalizeUrl = (url, pathStripRe) =>
+  'app://' + url.replace(/^file:\/\//, '').replace(pathStripRe, '')
```
Comment (enguerran) [inline src/drive/mobile/lib/reporter.js]: Can't we use the `url.replace` to _replace_ `file://` with `app://`:

```js
url.replace(/^file:\/\//, 'app://').replace(pathStripRe, '')
```

File: src/drive/mobile/lib/reporter.js [hunk 2]
```diff
@@ -6,11 +6,34 @@ export const ANALYTICS_URL =
...
+
+const normalizeData = data => {
+  const PATH_STRIP_RE = /^.*\/[^.]+(\.app|CodePush|.*(?=\/))/
+
+  if (data.culprit) {
+    data.culprit = normalizeUrl(data.culprit, PATH_STRIP_RE)
+  }
+  const stacktrace =
+    data.stacktrace || (data.exception && data.exception.values[0].stacktrace)
```
Comment (enguerran) [inline src/drive/mobile/lib/reporter.js]: You should check every part of the subproperty:

```js
data.stacktrace || (
  data.exception &&
  data.exception.values &&
  data.exception.values[0] &&
  data.exception.values[0].stacktrace
)
```

File: src/drive/mobile/lib/reporter.js [hunk 3]
```diff
@@ -6,11 +6,34 @@ export const ANALYTICS_URL =
...
+  if (stacktrace) {
+    stacktrace.frames.forEach(frame => {
+      if (frame.filename !== '[native code]') {
+        frame.filename = normalizeUrl(frame.filename, PATH_STRIP_RE)
+      }
+    })
```
Comment (enguerran) [inline src/drive/mobile/lib/reporter.js]: You should not hide side-effects:

```js
stacktrace.frames = stacktrace.frames.map(
  frame => frame.filename !== '[native code]' ?
  { ...frame, filename: normalizeUrl(frame.filename, PATH_STRIP_RE) } :
  frame
)
```

File: src/drive/mobile/lib/reporter.js [hunk 4]
```diff
@@ -6,11 +6,34 @@ export const ANALYTICS_URL =
...
+  }
+  return data
+}
```
Comment (enguerran) [inline src/drive/mobile/lib/reporter.js]: More generally, you should add test coverage on this small function.

**Comment by y-lohse** [comment]
Thanks for the review, all accurate comments… except for the need for tests maybe. 

They would take a lot of effort to set up, because the data object that the function receives comes from sentry, and it already seems to be doing some exception processing before that function. We'd mostly be testing raven.js which is not super useful.
Also this function is super isolated, so it would be quite hard to break. I don't see it evolving anytime soon either.
So yeah, I think adding tests here requires a lot of effort for very little gain.

All the other changes are done though 👍

**Comment by enguerran** [comment]
I thought that we need to test the function itself. It takes an object as a parameter and returns a new object. We should test this.

```js
const normalizeData = data => {
  const PATH_STRIP_RE = /^.*\/[^.]+(\.app|CodePush|.*(?=\/))/

  if (data.culprit) {
    data.culprit = normalizeUrl(data.culprit, PATH_STRIP_RE)
  }
  const stacktrace =
    data.stacktrace ||
    (data.exception &&
      data.exception.values &&
      data.exception.values[0] &&
      data.exception.values[0].stacktrace)

  if (stacktrace) {
    stacktrace.frames = stacktrace.frames.map(
      frame =>
        frame.filename !== '[native code]'
          ? { ...frame, filename: normalizeUrl(frame.filename, PATH_STRIP_RE) }
          : frame
    )
  }
  return data
}
```

And a test could be
```js
test("test normalizeData", () => {
  const data = { /* some data */ };
  const expectedData = { /* expected data */ };
  expect(normalizeData(result)).toEqual(expectedData);
})
```

What do you think?

**Comment by y-lohse** [comment]
So a fake object then? Sounds good to me. Still not sure it will be *that* useful, but t's low effort too so it's worth it!

**Pull request merged**